src/fidesops/service/connectors/saas_connector.py

```diff
@@ -255,6 +253,14 @@ def execute_prepared_request(
         client: AuthenticatedClient = self.create_client_from_request(saas_request)
         response: Response = client.send(prepared_request, saas_request.ignore_errors)
 
+        # replace errored response with empty dictionary if we are ignoring errors
+        if saas_request.ignore_errors and not response.ok:
+            logger.info(
+                f"Ignoring and clearing errored response with status code {response.status_code}."
+            )
+            response = Response()
+            response._content = b"{}"  # pylint: disable=W0212
+
```

Comment:
`execute_prepared_request` is getting pretty long now, and the behavior of replacing an errored response with an empty dictionary is not protected by a unit test, just an end-to-end sentry integration test.

I'd consider pulling this out into its own function then, the code comment can be the function docstring, and then a fake SaasRequest and Response can be created for unit testing just that function.

Reply:
nice suggestion @pattisdr, i'd also thought `execute_prepared_request` was getting rather long. i'll push an update here shortly

Reply:
https://github.com/ethyca/fidesops/pull/462/commits/be7e5e116b23af24ddac7629a1b1bc8c532a91b6 has a proposed implementation for the above. i also decided to pull out an `_unwrap_response_data` function while i was there for similar reasons, but please let me know if you don't think that's appropriate.

and i'm still getting a feel for python static/class/instance methods, so please let me know if you don't think that these should be static methods.